File: Submissions/#0359_div2/A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {

  int n, x;
  cin >> n >> x;

  long long answer = x;
  int count = 0;

  for (int i = 0; i < n; i++) {
    string k;
    cin >> k;
    int d;
    cin >> d;
    if (k == "+") {
      answer += d;
    } else {
      if (d <= answer) {
         answer -= d;
      } else {
        count++;
      }
    } 
  }

  cout << answer << " " << count << endl;

}

File: Submissions/#0359_div2/A/main.java
import java.util.*;

public class Main {

  public static void main (String[] args) {

    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();
    int x = sc.nextInt();
    
    long answer = x;
    int count = 0;

    for (int i = 0; i < n; i++) {
      String k = sc.next();
      int d = sc.nextInt();
      if (k.equals("+")) {
        answer += d;
      } else {
        if (d <= answer) {
           answer -= d;
        } else {
          count++;
        }
      } 
    }
    sc.close();

    System.out.println(answer + " " + count);

  }

}